GomokuClient/packages/gomoku-core/src/routes/game/join/$gameID.tsx

```diff
@@ -1,23 +1,29 @@
-import { QueryClient, useQuery } from "@tanstack/react-query";
+import { useQuery } from "@tanstack/react-query";
 import { createFileRoute } from "@tanstack/react-router";
 import { useEffect, useState } from "react";
 
 import type { SwaggerTypes } from "@/api";
 
 import { SwaggerServices } from "@/api";
-import { SignalRProvider, useAuthToken } from "@/context";
+import { useAuthToken } from "@/context";
 import JoinGame from "@/pages/JoinGame";
 import { getDefaultHeaders, typedSessionStorage } from "@/shared/lib/utils";
 import { LoadingOverlay } from "@/shared/ui/loading-overlay";
 import { toaster } from "@/shared/ui/toaster";
 
 const getGameHistory = async (
   gameID: SwaggerTypes.CreateGameResponse["gameId"],
+  jwtToken?: string,
 ) => {
-  const response = await SwaggerServices.getApiGameByGameIdHistory({
-    headers: getDefaultHeaders(),
-    path: { gameId: gameID },
-  });
+  const response = jwtToken
+    ? await SwaggerServices.getApiGameRegisteredByGameIdHistory({
+        headers: getDefaultHeaders(jwtToken),
+        path: { gameId: gameID },
+      })
+    : await SwaggerServices.getApiGameAnonymousByGameIdHistory({
+        headers: getDefaultHeaders(),
+        path: { gameId: gameID },
+      });
```

Comment:
I see, so we separate this logic and use a ternary operator to distinguish between anonymous and JWT-based users. I like the current approach, but I think it could be simplified across the app, as this ternary operator will be duplicated in many places. Let’s create a separate utility function for this:

```ts
async function fetchWithTokenFallback<T>(
  jwtToken: string | undefined,
  registeredPromise: () => Promise<T>,
  anonymousPromise: () => Promise<T>
): Promise<T> {
  return jwtToken ? registeredPromise() : anonymousPromise();
}
```

Reply:
Implementing such generic function is not that easy. As we have different responses, typescript cannot infer response type in place where function is used. Maybe this can be achieved with some really complex generic, but I spent already 40 minutes and was not able to do this 🤣

Reply:
I belive that should work, please let me know, if didn't, I will create then a seaprate PR on PR to handle this

```ts
export async function fetchWithAuthFallback<T, R = void>(
  jwtToken: string | undefined,
  authenticatedCall: (token: string) => Promise<T>,
  anonymousCall: (payload?: R) => Promise<T>,
  anonymousPayload?: R
): Promise<T> {
  return jwtToken 
    ? await authenticatedCall(jwtToken)
    : await anonymousCall(anonymousPayload);
}
```

Reply:
I pushed example here, please go to other sub functions to refactor to that! Huge thanks! https://github.com/ligomoku/gomoku/pull/234/commits/416d2b1514f27fcaa95a4d4cf9ae8d60723ab64d

Reply:
Done, wrapped that logic in function everywhere